realm/realm-library/src/objectServer/java/io/realm/SyncSession.java

```diff
@@ -873,7 +881,7 @@ protected AuthenticateResponse execute() {
             protected void onSuccess(AuthenticateResponse response) {
                 synchronized (SyncSession.this) {
                     if (!isClosed && !Thread.currentThread().isInterrupted() && !refreshTokenNetworkRequest.isCancelled()) {
```

Comment:
Not sure that I have the full overview of all execution paths, but seems like `refreshTokenNetworkRequest` could have been cleared in the meantime, so should be checked for null?

Reply:
Looking into it. This should be covered by `isClosed` which is set before setting the `refreshTokenNetworkRequest` to null. So it should be safe.